presto-hive-metastore/src/main/java/com/facebook/presto/hive/metastore/thrift/ThriftMetastoreUtil.java

```diff
@@ -822,7 +823,7 @@ public static Date toMetastoreDate(LocalDate date)
 
     public static Decimal toMetastoreDecimal(BigDecimal decimal)
     {
-        return new Decimal(ByteBuffer.wrap(decimal.unscaledValue().toByteArray()), (short) decimal.scale());
+        return new Decimal(Shorts.checkedCast(decimal.scale()), ByteBuffer.wrap(decimal.unscaledValue().toByteArray()));
```

Comment:
What's the reason for the changes for decimal?

Reply:
Constructor argument order is reversed in Hive 3. Trino also just switches the parameters around.